convert_fendl.py

```diff
@@ -41,90 +42,325 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
+        elif particle_details['file_type'] == "endf":
+            extraction_dir = endf_files_dir
 
-# ==============================================================================
-# GENERATE HDF5 LIBRARY -- NEUTRON FILES
+        for f in particle_details['files']:
+            # Extract files, the fendl release was compressed using type 9 zip format
+            # unfortunatly which is incompatible with the standard python zipfile library
+            # therefore the following system command is used
+            subprocess.call(['unzip', '-o', f, '-d', extraction_dir])
+        
+    os.chdir(cwd)
+    
+    if args.cleanup and download_path.exists():
+        rmtree(download_path)
 
-# Get a list of all ACE files, excluding files ending with _ which are old incorrect files kept in the release for backwards compatability
-neutron_files = [
-    f
-    for f in release_details[args.release]['neutron_files']
-    if not f.name.endswith('_') and not f.name.endswith('.xsd')
-]
+    # Seperate photon ENDF file for 2.1 release
+    if args.release == '2.1' and 'photon' in args.particles:
+        # In the 2.1 release, all the photon files are in one ENDF file so the
+        # from_endf method only reads the first nuclide. 
+        # This splits the file down into seperate ENDF files for later reading.
+        os.chdir(endf_files_dir)
+        split_endf('FENDLEP.DAT')
+        os.chdir(cwd)
 
-# Create output directory if it doesn't exist
-args.destination.mkdir(parents=True, exist_ok=True)
+# ==============================================================================
+# GENERATE HDF5 LIBRARY
 
 library = openmc.data.DataLibrary()
 
-for filename in sorted(neutron_files):
+warn_k39 = False    # Flag for K-39 error in FENDL-3.0 release
+
+for particle in args.particles:
+    # Create output directories if it doesn't exist
+    particle_destination = args.destination / particle
+    particle_destination.mkdir(parents=True, exist_ok=True)
+
+    particle_details = release_details[args.release][particle]
 
-    print(f'Converting: {filename}')
-    data = openmc.data.IncidentNeutron.from_ace(filename)
+    if particle == 'neutron':
+        # Get a list of all ACE files, excluding files ending with _ which are 
+        # old incorrect files kept in the release for backwards compatability
+        neutron_files = [
+            f
+            for f in particle_details['ace_files']
+            if not f.name.endswith('_') and not f.name.endswith('.xsd')
+        ]
+        for filename in sorted(neutron_files):
+            # Check for Inf values in K-39 ace file for FENDL-3.0
+            if args.release == '3.0' and filename.name == '19K_039.ace':
+                # Check for the error in case user has provided a fixed version.
+                if 'Inf' in open(filename, 'r').read():
+                    ace_error_warning = """
+                    WARNING: {} contains 'Inf' values within the XSS array which 
```

Comment:
If you use `warnings.warn` below, no need to add "WARNING" here
```suggestion
                    {} contains 'Inf' values within the XSS array which 
```